chain/src/params.rs

```diff
@@ -35,7 +37,7 @@ impl From<AssetInfo> for pb::AssetInfo {
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
This is necessary because of `ClientState`

Reply:
Since the `ChainParams` is the domain type for a proto type, we want to add Serde annotations so that the Serde format is derived from the proto formats, not from our internal representation:
 
```suggestion
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(try_from = pb::ChainParams, into = pb::ChainParams)]
```

This change won't work on its own, because currently the proto-derived `pb::ChainParams` type doesn't implement the Serde traits. To do that, we need to edit the `build.rs` in the `proto` crate to add
```
(".penumbra.chain.ChainParams", SERIALIZE)
```
to `TYPE_ATTRIBUTES` so that Prost will add the serde attributes when it generates the proto types.